packages/devtools_app/lib/src/screens/memory/memory_controller.dart

```diff
@@ -815,7 +818,16 @@ class MemoryController extends DisposableController
     // TODO(terry): Need an event on the controller for this too?
   }
 
-  void _handleConnectionStart(ServiceConnectionManager serviceManager) {
+  Future<void> _refreshShouldShowLeaksTab() async {
+    final isolateId = serviceManager.isolateManager.mainIsolate.value!.id!;
+    final Isolate isolate = await serviceManager.service!.getIsolate(isolateId);
+    _shouldShowLeaksTab.value =
+        isolate.extensionRPCs!.contains(leakTrackingExtensionName);
```

Comment:
use `serviceManager.serviceExtensionManager.hasServiceExtension(leakTrackingExtensionName)`
for this.

Example: https://github.com/flutter/devtools/blob/05d9d7482837bcce7659dbc0b1af7978500ba0c7/packages/devtools_app/lib/src/screens/inspector/inspector_controller.dart#L174